File: MicroHubManager/Backend/taskservices/controllers/taskController.js
import express from 'express'
import * as taskService from '../service/taskservice.js'



const router = express.Router();

router.post("/createtask",async(req,res)=>{
    res.json(await taskService.createTask(req.body,req.headers["token"]));
})



router.get("/getalltasks/:PAGE/:SIZE", async (req, res) => {
    const { PAGE, SIZE } = req.params;
    res.json(await taskService.getAllTasks(PAGE, SIZE, req.headers["token"]));
});


router.delete("/deletetask/:ID", async (req, res)=>{
    const {ID} = req.params;
    const response = await taskService.deleteTask(ID, req.headers["token"]);
    res.json(response);
});


export default router;

File: MicroHubManager/Backend/gateway/Controllers/taskController.py
# =======================================================================TASK CONTROLLER=====================================================
from fastapi import APIRouter,Header
from models.schemas import TaskSchema
import httpx


router = APIRouter(prefix="/taskservice")

SPRING_URL = "http://localhost:8002/"



@router.post("/createtask")
async def createtask(T: TaskSchema, token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.post(
            SPRING_URL + "task/createtask",
            json=T.model_dump(),
            headers={"Token": token}
        )
    return response.json()

@router.get("/getalltasks/{PAGE}/{SIZE}")
async def getAllTasks(PAGE: int, SIZE:int, token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.get(
            SPRING_URL + f"task/getalltasks/{PAGE}/{SIZE}",
            headers = {"Token": token}
        )
    return response.json()


@router.delete("/deletetask/{ID}")
async def deleteFunction(ID: str, token: str = Header(...)):
    async with httpx.AsyncClient() as client:
        response = await client.delete(
            SPRING_URL + f"task/deletetask/{ID}",
            headers = {"Token": token}
        )
    return response.json()